File: src/main/java/org/example/model/dto/CourseDto.java
package org.example.model.dto;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CourseDto {
    private Long id;
    private String code;
    private String name;
    private String description;
    private Integer ects;

    private Boolean mandatory;
    private Integer enrollmentLimit;
    private Integer semester;
    private Boolean active;

    private Long lecturerId;
    private Long studyProgramId;
    private Long academicYearId;
}

File: frontend/src/data/dto/course.dto.ts
export type CreateCourseRequestDto = {
  code: string;
  name: string;
  description?: string | null;
  ects: number;
  semester: number;
  mandatory: boolean;
  enrollmentLimit?: number | null;
  lecturerId?: number | null;
  studyProgramId?: number | null;
  academicYearId?: number | null;
  active: boolean;
  prerequisiteCourseIds?: number[];
};

export type CourseDto = {
  id: number;
  code: string;
  name: string;
  description: string | null;
  ects: number;
  semester: number;
  mandatory: boolean;
  active: boolean;
  enrollmentLimit: number | null;
  studyProgramId: number | null;
  academicYearId: number | null;
  lecturerId: number | null;
};

export type UpdateCourseRequestDto = {
  any: any;
};
